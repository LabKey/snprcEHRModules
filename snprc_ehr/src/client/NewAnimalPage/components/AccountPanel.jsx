import React from 'react';
import Select from 'react-select';


export default class AccountPanel extends React.Component {

    handleAccountChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('animalAccount', validatedOption);
    }
    handleColonyChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('colony', validatedOption);
    }
    handleOwnerInstitutionChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('ownerInstitution', validatedOption);
    }
    handleIacucChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('iacuc', validatedOption);
    }
    handlePedigreeChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('pedigree', validatedOption);
    }    
    handleResponsibleInstitutionChange = option => {
        const validatedOption = this.validate(option);
        this.props.handleDataChange('responsibleInstitution', validatedOption);
    }
    validate = option => {
        return ( {
            ...option,
            hasError: false
        })
    }
    render() {
        let {animalAccount, colony, pedigree, iacuc, responsibleInstitution, ownerInstitution } = this.props.newAnimalData;
        return (
            <div className="wizard-panel__rows">
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >Account</label>
                        <Select
                                defaultValue={animalAccount}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.accountList}
                                onChange={this.handleAccountChange}
                                placeholder="Select Account"
                                isDisabled={this.props.disabled}
                                id="account-select"
                        />
                    </div>
                </div>
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >Colony</label>
                        <Select 
                                defaultValue={colony}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.colonyList}
                                onChange={this.handleColonyChange}
                                placeholder="Select Colony"
                                isDisabled={this.props.disabled}
                                id="colony-select"
                        />
                    </div>
                </div>
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >IACUC</label>
                        <Select 
                                defaultValue={iacuc}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.iacucList}
                                onChange={this.handleIacucChange}
                                placeholder="Select IACUC"
                                isDisabled={this.props.disabled}
                                id="iacuc-select"
                        />
                    </div>
                </div>
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >Pedigree</label>
                        <Select 
                                defaultValue={pedigree}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.pedigreeList}
                                onChange={this.handlePedigreeChange}
                                placeholder="Select Pedigree"
                                isDisabled={this.props.disabled}
                                id="pedigree-select"
                        />
                    </div>
                </div>
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >Owner</label>
                        <Select 
                                defaultValue={ownerInstitution}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.institutionList}
                                onChange={this.handleOwnerInstitutionChange}
                                placeholder="Select Owner"
                                isDisabled={this.props.disabled}
                                id="owner-select"
                        />
                    </div>
                </div>
                <div className='wizard-panel__row' >
                    <div className='wizard-panel__col'>
                        <label className="field-label" >Responsible Institution</label>
                        <Select 
                                defaultValue={responsibleInstitution}
                                className="shared-dropdown"
                                classNamePrefix="shared-select"
                                options={this.props.institutionList}
                                onChange={this.handleResponsibleInstitutionChange}
                                placeholder="Select Responsible Institution"
                                isDisabled={this.props.disabled}
                                id="responsible-institution-select"
                        />
                    </div>
                </div>                
            </div> 
        )
    }
}
