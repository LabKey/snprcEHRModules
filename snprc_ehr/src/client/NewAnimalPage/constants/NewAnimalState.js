import moment from 'moment';

const NewAnimalState = () => {

    return  ({
        currentPanel: 1,
        selectedOption: undefined,
        newAnimalData: {
            id: undefined,
            birthDate: { date: moment() },
            acquisitionType: undefined,
            acqDate: { date: moment() },
            gender: undefined,
            sire: undefined,
            dam: undefined,
            species: undefined,
            colony: undefined,
            animalAccount: undefined,
            ownerInstitution: undefined,
            responsibleInstitution: undefined,
            room: undefined,
            cage: { value: undefined },
            diet: undefined,
            pedigree: undefined,
            iacuc: undefined
        },
        speciesList: [],
        acquisitionTypeList: [],
        potentialDamList: [],
        potentialSireList: [],
        locationList: [],
        accountList: [],
        colonyList: [],
        institutionList: [],
        iacucList: [],
        pedigreeList: [],
        isLoading: true,
        hasError: false,
        preventNext: true,
        saveOk: true,
        showSaveModal: false,
        showCancelModal: false,
        showSpeciesChangeModal: false

    });
}

export default NewAnimalState;