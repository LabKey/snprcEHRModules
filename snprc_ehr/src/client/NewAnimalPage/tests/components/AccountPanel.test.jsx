import React from 'react'
import { shallow } from 'enzyme'
import AccountPanel from '../../components/AccountPanel'
import NewAnimalState from '../../constants/NewAnimalState'
import lists from '../fixtures/lists'

test('Should render the AccountPanel', () => {
    const newAnimalData = {
      ...(NewAnimalState().newAnimalData),
      acquisitionType: { id: 0, value: 1, label: '1 - Colony-born, Vaginal delivery (at TBRI)', Category: 'Birth', SortOrder: 1 }
    }

    const wrapper = shallow(
      <AccountPanel
        accountList={ lists.accountList }
        colonyList={ lists.colonyList }
        disabled={ false }
        handleDataChange={ () => { } }
        iacucList={ lists.iacucList }
        institutionList={ lists.institutionList }
        newAnimalData={ newAnimalData }
        pedigreeList={ lists.pedigreeList }
        preventNext={ () => { return false } }
      />
    )
    expect(wrapper).toMatchSnapshot()
})
