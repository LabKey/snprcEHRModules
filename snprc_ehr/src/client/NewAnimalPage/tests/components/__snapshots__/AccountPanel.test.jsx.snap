// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the AccountPanel 1`] = `
<Fragment>
  <div
    className="wizard-panel__rows"
    id="account-panel"
  >
    <div
      className="wizard-panel__row"
      id="account-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Account
        </label>
        <StateManager
          autoFocus={true}
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="account-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": 1,
                "label": "0063-010-99 - TEST Blah, Combination therapy for Blah",
                "value": "0063-010-99",
              },
              Object {
                "id": 0,
                "label": "0060-014-99 - TEST PC, Natural resistance of baboon cells to Blah infection",
                "value": "0060-014-00",
              },
            ]
          }
          placeholder="Select Account"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
      id="colony-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Colony
        </label>
        <StateManager
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="colony-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "category": 0,
                "id": 0,
                "label": "test Colony 1",
                "species": "PC",
                "value": 1,
              },
              Object {
                "category": 0,
                "id": 1,
                "label": "test Colony 2",
                "species": "PC",
                "value": 2,
              },
            ]
          }
          placeholder="Select Colony"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
      id="iacuc-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          IACUC
        </label>
        <StateManager
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="iacuc-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": 0,
                "label": "111PC - Test",
                "maxAnimals": 0,
                "species": "PC",
                "value": "111PC",
              },
              Object {
                "id": 1,
                "label": "222PC - Test",
                "maxAnimals": 0,
                "species": "PC",
                "value": "222PC",
              },
            ]
          }
          placeholder="Select IACUC"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
      id="pedigree-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Pedigree
        </label>
        <StateManager
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="pedigree-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "category": 2,
                "id": 0,
                "label": "Baboon 1",
                "species": "PC",
                "value": 31,
              },
              Object {
                "category": 2,
                "id": 1,
                "label": "Baboon 2",
                "species": "PC",
                "value": 32,
              },
            ]
          }
          placeholder="Select Pedigree"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
      id="owner-institution-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Owner
        </label>
        <StateManager
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="owner-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": 0,
                "label": "TxBiomed - Texas Biomedical Research Institute",
                "value": 1,
              },
              Object {
                "id": 1,
                "label": "Mexico Biomed- Mexico Biomedical Research Institute",
                "value": 2,
              },
            ]
          }
          placeholder="Select Owner"
        />
      </div>
    </div>
    <div
      className="wizard-panel__row"
      id="responsible-institution-row"
    >
      <div
        className="wizard-panel__col"
      >
        <label
          className="field-label"
        >
          Responsible Institution
        </label>
        <StateManager
          className="shared-dropdown"
          classNamePrefix="shared-select"
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          id="responsible-institution-select"
          isClearable={true}
          isDisabled={false}
          isLoading={false}
          onChange={[Function]}
          options={
            Array [
              Object {
                "id": 0,
                "label": "TxBiomed - Texas Biomedical Research Institute",
                "value": 1,
              },
              Object {
                "id": 1,
                "label": "Mexico Biomed- Mexico Biomedical Research Institute",
                "value": 2,
              },
            ]
          }
          placeholder="Select Responsible Institution"
        />
      </div>
    </div>
  </div>
  <InfoPanel
    messages={
      Array [
        Object {
          "colName": "Account",
          "propTest": true,
        },
        Object {
          "colName": "Colony",
          "propTest": true,
        },
        Object {
          "colName": "IACUC",
          "propTest": true,
        },
        Object {
          "colName": "Pedigree",
          "propTest": true,
        },
        Object {
          "colName": "Owner",
          "propTest": true,
        },
        Object {
          "colName": "Responsible Institution",
          "propTest": true,
        },
      ]
    }
  />
</Fragment>
`;
